packages/hadron-document/src/object-generator.ts

```diff
@@ -436,28 +509,29 @@ export class ObjectGenerator {
     const query: any = {};
     if (
       originalFields.some(({ path }) =>
-        path.some((key) => key.includes('.') || key.startsWith('$'))
+        path.some(({ key }) => key.includes('.') || key.startsWith('$'))
       )
     ) {
       // Some of the keys in this query are only accesible via $getField,
       // which was introduced in MongoDB 5.0.
+      const equalityMatches: any[] = [];
       for (const { path, value } of originalFields) {
         const getFieldExpr = ObjectGenerator.createGetFieldExpr(path);
-        const matchExpr =
+        equalityMatches.push(
           value !== DoesNotExist
-            ? {
-                $eq: [getFieldExpr, { $literal: value }],
-              }
-            : {
-                $eq: [{ $type: getFieldExpr }, 'missing'],
-              };
-        query.$expr ??= { $and: [] };
-        query.$expr.$and.push(matchExpr);
+            ? { $eq: [getFieldExpr, { $literal: value }] }
+            : { $eq: [{ $type: getFieldExpr }, 'missing'] }
+        );
+      }
+      if (equalityMatches.length === 1) {
+        query.$expr = equalityMatches[0];
+      } else if (equalityMatches.length > 1) {
+        query.$expr = { $and: equalityMatches };
```

Comment:
This part of the change is also a little independent, and only makes it so that `$and: [singleEntry]` is reduced to `singleEntry`, with the main purpose being making the test files a *little* more readable